cmd/encode-openapi-spec/main.go

```diff
@@ -0,0 +1,109 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"flag"
+	"fmt"
+	"go/format"
+	"html/template"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type templateInput struct {
+	Version int
+	Format  string
+	Spec    string
+}
+
+const tmpl = `package openapi
+
+// OpenAPIV{{.Version}}Spec contains a gzip-compressed {{.Format}} file specifying the Temporal HTTP API
+var OpenAPIV{{.Version}}Spec = {{.Spec}}`
```

Comment:
```suggestion
var OpenAPIV{{.Version}}{{.Lang}}Spec = {{.Spec}}`
```

Sorry for not making this observation earlier, but I think we need to put `YAML` or `JSON` in the name. I was under the impression at least one of those OpenAPI versions supports both.

Reply:
Ah good catch, I'll add that